app/tests/testsv2/test_user_signup.py

```diff
@@ -0,0 +1,32 @@
+from . import BaseClass
+
+
+class UserTestCase(BaseClass):
+    """This class represents the user test case"""
+    def test_signup(self):
+        """Test API can signup (POST request)"""
+        res = self.client().post("api/v2/auth/signup", json=self.user)
+        self.assertEqual(res.status_code, 201)
+        self.assertIn("chirchir Kibet", str(res.data))
+
+    def test_signup_fail(self):
+        """Test API cannot take null (POST request)"""
+        res = self.client().post("api/v2/auth/signup", data=self.user)
+        self.assertEqual(res.status_code, 400)
+        self.assertIn("provide all fields", str(res.data))
```

Comment:
I don't see why the test `test_signup ` should pass and this test passes. They send the same request with the same data. Moreso, `self.user` in this case has all the fields provided, is it?

Reply:
It is passing because I'm not sending a json file I'm sending data

Reply:
On my view im receiving a json data(api.payload) so if i send data it will not be a json which im expecting to recieve